### PR TITLE
markdown editor v6

### PR DESCRIPTION
**Pull request by burtonator**
Don't do a full QA of this PR yet.  I'm still fixing a few more Safari + Mobile issues.  

I'm going to start testing the integration into the site next and we can start being more picky then as some of the functionality might change.  

The biggest change here is fixing Safari on iOS / iPad OS 18.1 and solving the keyboard detection issue.

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/9709

## Description of Changes

Fixed the following issues.

-  Focus seems to be inconsistent with the keyboard use so I need to double check everything and audit it.
- Make sure the secondary buttons focus properly. I don't them all of them are.
- selecting the 'p' block style in Safari on iPad causes the editor to lose focus and it goes to 'body'. This doesn't happen for h1, etc.
- paragraph selection DOES cause focus to be lost on mobile
- test safari + ipad
-  keyboard does NOT become active on Safari + iPad
-  The padding for the button on mobile button bar on iOS seems wrong

## Test Plan
- none just yet.

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/FormatButton.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ export const FormatButton = (props: HeadingButtonProps) => {
       content={`Change to ${formatName}`}
       renderTrigger={(handleInteraction) => (
         <CWIconButton
-          className={clsx({ FormatButtonActive: active })}
+          className={clsx({ FormatButton: true, FormatButtonActive: active })}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/FormatButton.tsx:46]:
```suggestion
          className={clsx('FormatButton', { FormatButtonActive: active })}
```

File: packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/FormatButton.scss [hunk 2]
```diff
@@ -1,5 +1,8 @@
 @import '../../../../../styles/shared';
 
+.FormatButton {
+}
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/FormatButton.scss:4]: remove empty class from this file

File: packages/commonwealth/client/scripts/views/components/MarkdownEditor/useMobileKeyboardResizeHandler.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { MarkdownEditorMode } from 'views/components/MarkdownEditor/MarkdownEditor';
+
+/**
+ * Do the actual keyboard resize now.
+ */
+export function resizeRootElementForKeyboard() {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/components/MarkdownEditor/useMobileKeyboardResizeHandler.ts:7]: you don't need export

**Review by masvelio — APPROVED** [review]
some small comments

**Review by Israellund — APPROVED** [review]

File: packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/BlockSelectorButton.tsx [hunk 4]
```diff
@@ -45,12 +45,11 @@ export const BlockSelectorButton = (props: BlockSelectorButtonProps) => {
   return (
     <ClickAwayListener onClickAway={handleClickAway}>
       <div className="BlockSelectorButton">
-        <button onClick={handleClick}>
-          <div className="DropdownIndicator">
-            {iconName && <CWIcon iconName={iconName} />}
-            {!iconName && <PlaceholderIcon />}
-            <CWIcon iconName="caretDown" iconSize="xs" />
-          </div>
+        <button onClick={handleClick} style={{ display: 'flex' }}>
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/MarkdownEditor/toolbars/BlockSelectorButton.tsx:48]: use css class plz

**Review by mzparacha — APPROVED** [review]

**Pull request merged**